tests/test_hardware_parity.py

```
@@ -0,0 +1,898 @@
+        self.assertTrue(errors)
+
+    def test_non_dict_deployment(self):
+        self._assert_reports(
+            lambda record: record["oracle"].__setitem__("deployment", "tampered")
+        )
+
+    def test_truthy_non_dict_oracle(self):
+        self._assert_reports(lambda record: record.__setitem__("oracle", ["x"]))
+
+    def test_ragged_spike_grid(self):
+        def mutate(record):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `mutate`

Add return type annotation: `None`

(ANN202)